php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1402 @@
+  }
+
+  public InternalSyntaxToken EOS() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.EOS)
+        .is(b.firstOf(b.token(PHPPunctuator.SEMICOLON), b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  /**
+   * [ END ] Statement
+   */
+
+  public ExpressionTree EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EXPRESSION)
+      .is(f.expression(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  /**
+   * [ START ] Expression
+   */
+
+  // FIXME Fake implementation
+  public ExpressionTree UNARY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.UNARY_EXPR).is(
+      b.firstOf(
+        COMMON_SCALAR(),
+        EXPRESSION()));
+  }
+
+  public ExpressionTree CONCATENATION_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.CONCATENATION_EXPR).is(
+      f.concatenationExpr(
+        UNARY_EXPR(),
+        b.zeroOrMore(f.newTuple60(
+          b.token(DOT),
+          UNARY_EXPR()))));
+  }
+
+  public ExpressionTree MULTIPLICATIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.MULTIPLICATIVE_EXPR).is(
+      f.multiplicativeExpr(
+        CONCATENATION_EXPR(),
+        b.zeroOrMore(f.newTuple61(
+          b.firstOf(b.token(STAR), b.token(DIV), b.token(MOD)),
+          CONCATENATION_EXPR()))));
+  }
+
+  public ExpressionTree ADDITIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ADDITIVE_EXPR).is(
+      f.additiveExpr(
+        MULTIPLICATIVE_EXPR(),
+        b.zeroOrMore(f.newTuple62(
+          b.firstOf(b.token(PLUS), b.token(MINUS)),
+          MULTIPLICATIVE_EXPR()))));
+  }
+
+  public ExpressionTree SHIFT_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SHIFT_EXPR).is(
+      f.shiftExpr(
+        ADDITIVE_EXPR(),
+        b.zeroOrMore(f.newTuple63(
+          b.firstOf(b.token(SL), b.token(SR)),
+          ADDITIVE_EXPR()))));
+  }
+
+  public ExpressionTree RELATIONAL_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.RELATIONAL_EXPR).is(
+      f.relationalExpr(
+        SHIFT_EXPR(),
+        b.zeroOrMore(f.newTuple64(
+          b.firstOf(b.token(LE), b.token(GE), b.token(LT), b.token(GT)),
+          SHIFT_EXPR()))));
+  }
+
+  public ExpressionTree EQUALITY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EQUALITY_EXPR).is(
+      f.equalityExpr(
+        RELATIONAL_EXPR(),
+        b.zeroOrMore(f.newTuple65(
+          b.firstOf(b.token(NOTEQUAL2), b.token(NOTEQUAL), b.token(EQUAL2), b.token(EQUAL), b.token(NOTEQUALBIS)),
+          RELATIONAL_EXPR()))));
+  }
+
+  public ExpressionTree BITEWISE_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.BITEWISE_AND_EXPR).is(
+      f.bitwiseAndExpr(
+        EQUALITY_EXPR(),
+        b.zeroOrMore(f.newTuple66(
+          b.token(PHPPunctuator.AMPERSAND),
+          EQUALITY_EXPR()))));
+  }
+
+  public ExpressionTree BITEWISE_XOR_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)